src/test/plugins/mvcc_delete_plugin_test.cpp

```diff
@@ -0,0 +1,210 @@
+
+    const auto& where_scan =
+        std::make_shared<TableScan>(get_table, expression_functional::greater_than_(column_a, val));
+    where_scan->set_transaction_context(transaction_context);
+    where_scan->execute();
+
+    const auto& update = std::make_shared<Update>(name, where_scan, where_scan);
+    update->set_transaction_context(transaction_context);
+    update->execute();
+
+    transaction_context->commit();
+  }
+
+  void load_plugin() {
+    auto& pm = PluginManager::get();
+    pm.load_plugin(build_dylib_path("libMvccDeletePlugin"));
+  }
+
+  void unload_plugin() {
+    auto& pm = PluginManager::get();
+    pm.unload_plugin("MvccDeletePlugin");
+  }
+};
+
+TEST_F(MvccDeletePluginTest, LoadUnloadPlugin) {
+  load_plugin();
+  unload_plugin();
+}
+
+class MvccDeletePluginCoreTest : public BaseTest {
+ public:
+  static void SetUpTestCase() { _column_a = pqp_column_(ColumnID{0}, DataType::Int, false, "a"); }
+
+  void SetUp() override {}  // managed by each test individually
+
+  void TearDown() override { StorageManager::reset(); }
+
+ protected:
+  void _increment_all_values_by_one() {
+    auto transaction_context = TransactionManager::get().new_transaction_context();
+    // GetTable
+    auto get_table = std::make_shared<GetTable>(_table_name);
+    get_table->set_transaction_context(transaction_context);
+    get_table->execute();
+
+    // Validate
+    auto validate_table = std::make_shared<Validate>(get_table);
+    validate_table->set_transaction_context(transaction_context);
+    validate_table->execute();
+
+    // Update
+    auto update_expressions = expression_vector(add_(_column_a, 1));
+    auto updated_values_projection = std::make_shared<Projection>(validate_table, update_expressions);
+    updated_values_projection->execute();
+    auto update_table = std::make_shared<Update>(_table_name, validate_table, updated_values_projection);
+    update_table->set_transaction_context(transaction_context);
+    update_table->execute();
+
+    transaction_context->commit();
+  }
+  static bool _delete_chunk_logically(const std::string& table_name, ChunkID chunk_id) {
+    return MvccDeletePlugin::_delete_chunk_logically(table_name, chunk_id);
+  }
+  static bool _delete_chunk_physically(const std::string& table_name, ChunkID chunk_id) {
+    return MvccDeletePlugin::_delete_chunk_physically(table_name, chunk_id);
+  }
+  static int _get_int_value_from_table(const std::shared_ptr<const Table> table, const ChunkID chunk_id,
+                                       const ColumnID column_id, const ChunkOffset chunk_offset) {
+    const auto segment = table->get_chunk(chunk_id)->get_segment(column_id);
+    const auto& value_alltype = static_cast<const AllTypeVariant&>((*segment)[chunk_offset]);
+    return boost::lexical_cast<int>(value_alltype);
+  }
+
+  std::string _table_name{"mvccTestTable"};
+  inline static std::shared_ptr<AbstractExpression> _column_a;
+};
+
+TEST_F(MvccDeletePluginCoreTest, LogicalDelete) {
```

Comment:
As these tests are quite complex, please add some comments on what you actually what to test.